### PR TITLE
feat: Implement Chat Memory Manager node

### PR DESCRIPTION
**Pull request by OlegIvaniv**
## Summary
This PR implements **Chat Memory Manager** to allow for retrieving and mutating of a chat memory in the main flow.
It replaces the old `Chat Messages Retriever` and adds two additional operations: in(up)sert  and delete.

### CODE REVIEW

File: packages/@n8n/nodes-langchain/nodes/memory/MemoryManager/MemoryManager.node.ts [hunk 1]
```diff
@@ -0,0 +1,329 @@
...
+
+				if (insertMode === 'override') {
+					await memory.chatHistory.clear();
+				}
+
+				for (const message of messagesToInsert) {
+					const MessageClass = new templateMapper[message.type](message.message);
+
+					if (message.hideFromUI) {
+						MessageClass.additional_kwargs.hideFromUI = true;
```
Comment (michael-radency) [inline packages/@n8n/nodes-langchain/nodes/memory/MemoryManager/MemoryManager.node.ts:299]: would `MessageClass` always have `additional_kwargs`? should we consider case when it's `undefined`?

Comment (OlegIvaniv) [reply packages/@n8n/nodes-langchain/nodes/memory/MemoryManager/MemoryManager.node.ts:299]: It should, as per Langchain types, this propert is not optional and `NonNullable`:

```ts
/**
 * Exclude null and undefined from T
 */
type NonNullable<T> = T & {};

    /** Additional keyword arguments */
    additional_kwargs: NonNullable<BaseMessageFields["additional_kwargs"]>;
```

**Review by michael-radency — APPROVED** [review]
LGTM

**Comment by janober** [comment]
Got released with [`n8n@1.25.0`](https://github.com/n8n-io/n8n/releases/tag/n8n%401.25.0)

**Pull request merged**